src/cloud_common/protocols/unit/desired_status.py

```diff
@@ -0,0 +1,431 @@
+        str,
+        Field(
+            description='Unique name of the certificate chain',
+        ),
+    ]
+
+    fingerprints: Annotated[
+        list[str],
+        Field(
+            description='fingerprint list of the certificates included in the chain',
+        ),
+    ]
+
+
+class AosSign(BaseModel):
+    """Aos sign information."""
+
+    chain_name: Annotated[
+        str,
+        Field(
+            alias='chainName',
+            description='chain name from the list of `certificateChains`',
+        ),
+    ]
+
+    alg: Annotated[
+        Literal['RSA/SHA256', 'EC/SHA256'],
+        Field(
+            description='used algorithms for signing in the form `alg/hash`',
+        ),
+    ]
+
+    value: Annotated[  # noqa: WPS110
+        Base64Bytes,
+        Field(
+            description='value of the signature',
+        ),
+    ]
+
+    timestamp: Annotated[
+        datetime,
+        Field(
+            alias='trustedTimestamp',
+            description='Timestamp of the signature in ISO8601 format',
+        ),
+    ]
+
+    ocsp_values: Annotated[
+        list[Base64Bytes],
+        Field(
+            alias='ocspValues',
+            default=None,
+            description='OCSP values of the signature',
+        ),
+    ]
+
+
+class AosTimeSlot(BaseModel):
+    """Timetable time slot."""
+
+    start: Annotated[
+        time,
+        Field(
+            description='Start time in form `HH:MM[:SS]`.',
+        ),
+    ]
+
+    end: Annotated[
+        time,
+        Field(
+            description='End time in form `HH:MM[:SS]`.',
+        ),
+    ]
+
+
+class AosTimetableItem(BaseModel):
+    """
+    Timetable signe record.
+
+    Represent one entry of the timetable in form
+    `day of week`: [start time:end time]
+    """
+
+    day_of_week: Annotated[
+        int,
+        Field(
+            alias='dayOfWeek',
+            description='day of the week: Monday [1] ... Sunday [7]',
+        ),
+    ]
+
+    time_slots: Annotated[
+        list[AosTimeSlot],
+        Field(
+            alias='timeSlots',
+            min_items=1,
+            description='List of the time slots for the timetable.',
+        ),
+    ]
+
+
+class AosScheduleRule(BaseModel):
+    """Aos schedule rule."""
+
+    ttl: Annotated[
+        int,
+        Field(
+            description='TTL of the rule in seconds.',
+        ),
+    ]
+
+    type: Annotated[
+        Literal['force', 'trigger', 'timetable'],
+        Field(
+            description='Type of the Schedule rule.',
+        ),
+    ]
+
+    timetable: Annotated[
+        list[AosTimetableItem],
+        Field(
+            default=None,
+            description='Timetable when rule must work (only when the type is `timetable`).',
+        ),
+    ]
+
+
+class AosDesiredComponentInfo(BaseModel):
+    """Component info sent from the AosEdge Cloud."""
+
+    version: TypeVersionMandatory
+    type: TypeComponentType
+    annotations: TypeComponentAnnotationsOptional
+
```

Comment:
We agreed to add `id` field` as well. As we may have few components with same type but different versions.

Reply:
done